### PR TITLE
fix: Ichi-getSpecificName() returns symbol

### PR DESCRIPTION
**Pull request by hcrypto7**

### CODE REVIEW

**Review by a17 — APPROVED** [review]

**Review by a17 — CHANGES_REQUESTED** [review]
allowed assets symbol

File: src/strategies/IchiRetroMerklFarmStrategy.sol [hunk 1]
```diff
@@ -270,8 +270,12 @@ contract IchiRetroMerklFarmStrategy is LPStrategyBase, FarmingStrategyBase {
     /// @inheritdoc IStrategy
     function getSpecificName() external view override returns (string memory, bool) {
         IFactory.Farm memory farm = _getFarm();
-        string memory shortAddr = IRMFLib.shortAddress(farm.addresses[0]);
-        return (shortAddr, true);
+        IICHIVault _ivault = IICHIVault(farm.addresses[0]);
+        address allowedToken;
```
Comment (a17) [inline src/strategies/IchiRetroMerklFarmStrategy.sol:274]: write something like
```solidity
address allowedToken = _ivault.allowToken0() ? _ivault.token0() : _ivault.token1();
```
to save gas and reduce branches

File: src/strategies/IchiQuickSwapMerklFarmStrategy.sol [hunk 2]
```diff
@@ -162,8 +162,12 @@ contract IchiQuickSwapMerklFarmStrategy is LPStrategyBase, FarmingStrategyBase {
     /// @inheritdoc IStrategy
     function getSpecificName() external view override returns (string memory, bool) {
         IFactory.Farm memory farm = _getFarm();
-        string memory shortAddr = IQMFLib.shortAddress(farm.addresses[0]);
-        return (shortAddr, true);
+        IICHIVault _ivault = IICHIVault(farm.addresses[0]);
+        address allowedToken;
```
Comment (a17) [inline src/strategies/IchiQuickSwapMerklFarmStrategy.sol:166]: write something like
```solidity
address allowedToken = _ivault.allowToken0() ? _ivault.token0() : _ivault.token1();
```
to save gas and reduce branches

**Review by a17 — CHANGES_REQUESTED** [review]

**Review by hcrypto7 — COMMENTED** [review]
fixed problem.

File: src/strategies/IchiQuickSwapMerklFarmStrategy.sol (hunk 2 above)
Comment (hcrypto7) [reply src/strategies/IchiQuickSwapMerklFarmStrategy.sol]: thank you. I fixed issues.

**Review by a17 — APPROVED** [review]

**Pull request merged**